RecoBTag/DeepFlavour/plugins/DeepFlavourTagInfoProducer.cc

```diff
@@ -0,0 +1,341 @@
+    btagbtvdeep::DeepFlavourFeatures features;
+
+    // reco jet reference (use as much as possible)
+    const auto & jet = jets->at(jet_n);
+    // dynamical castoting to pointers, null if not possible
+    const auto * pf_jet = dynamic_cast<const reco::PFJet *>(&jet);
+    edm::RefToBase<reco::Jet> jet_ref(jets, jet_n);
+    // TagInfoCollection not in an associative container so search for matchs
+    const edm::View<reco::ShallowTagInfo> & taginfos = *shallow_tag_infos;
+    edm::Ptr<reco::ShallowTagInfo> match;
+    // Try first by 'same index'
+    if ((jet_n < taginfos.size()) && (taginfos[jet_n].jet() == jet_ref)) {
+        match = taginfos.ptrAt(jet_n);
+    } else {
+      // otherwise fail back to a simple search
+      for (auto itTI = taginfos.begin(), edTI = taginfos.end(); itTI != edTI; ++itTI) {
+        if (itTI->jet() == jet_ref) { match = taginfos.ptrAt( itTI - taginfos.begin() ); break; }
+      }
+    }
+    reco::ShallowTagInfo tag_info;
+    if (match.isNonnull()) {
+      tag_info = *match; 
+    } // will be default values otherwise
+
+    // fill basic jet features
+    btagbtvdeep::JetConverter::JetToFeatures(jet, features.jet_features);
+
+    // fill number of pv
+    features.npv = vtxs->size();
+
+    // fill features from ShallowTagInfo
+    const auto & tag_info_vars = tag_info.taggingVariables();
+    btagbtvdeep::BTagConverter::BTagToFeatures(tag_info_vars, features.tag_info_features);
+
+    // copy which will be sorted
+    auto svs_sorted = *svs;     
+    // sort by dxy
+    std::sort(svs_sorted.begin(), svs_sorted.end(),
+              [&pv](const auto & sva, const auto &svb)
+              { return btagbtvdeep::sv_vertex_comparator(sva, svb, pv); });
+    // fill features from secondary vertices
+    for (const auto & sv : svs_sorted) {
+      if (reco::deltaR(sv, jet) > jet_radius_) continue;
+      else {
+        features.sv_features.emplace_back();
+        // in C++17 could just get from emplace_back output
+        auto & sv_features = features.sv_features.back();
+        btagbtvdeep::SVConverter::SVToFeatures(sv, pv, jet, sv_features);
+      }
+    }
+
+    // stuff required for dealing with pf candidates 
+    math::XYZVector jet_dir = jet.momentum().Unit();
+    GlobalVector jet_ref_track_dir(jet.px(),
+                                   jet.py(),
+                                   jet.pz());
+
+    std::vector<btagbtvdeep::SortingClass<size_t> > c_sorted, n_sorted;
+
+    // to cache the TrackInfo
+    std::map<unsigned int, btagbtvdeep::TrackInfoBuilder> trackinfos;
+
+    // unsorted reference to sv
+    const auto & svs_unsorted = *svs;     
+    // fill collection, from DeepTNtuples plus some styling
+    for (unsigned int i = 0; i <  jet.numberOfDaughters(); i++){
+        auto cand = jet.daughter(i);
+        if(cand){
+          // candidates under 950MeV are not considered
+          // might change if we use also white-listing
+          if (cand->pt()<0.95) continue; 
```

Comment:
should this be configurable?

Reply:
We could make it configurable, but I guess the only values would be really no filter or this 0.95 GeV filter.

Reply:
configurable from https://github.com/cms-sw/cmssw/pull/19893/commits/74853fa8013be8de58f3fa7f011fea1cd51f0358